### ISSUE TITLE
Contrast issue with Google Docs in document Editing mode 

### ISSUE THREAD

**Issue by arigit** [DmitriK/darkContrast#20]
There is a long-standing little annoyance when using google docs that is missed by darkContrast

To reproduce:
In your docs.google.com, create a new blank document

Notice that when the editor opens up, above the Menu Bar (the bar with File | Edit | View etc), the document title ("Untitled Document", by default, but can be changed by clicking) appears inside a dark-grey box with dark (nearly black) letters, i.e. it can't be read.

It would be nice if darkContrast could change the background to white as the rest of the page.

Tested on Fedora 25 w/ Dark Adwaita theme / Firefox 53  w/Compact Dark theme

Thanks for this great extension

**Comment by DmitriK** [comment]
Ok, from what I can tell, Google has a div containing the title text that you normally see. Just below that, is an input with the text intentionally set to white, and they are assuming the default input color is white (of course), which causes the problem. The text is set to a proper color when the input is in focus, but returns to white when out of focus. This is a super hacky way to do a simple input element.

This may be difficult to fix in a generic way. Google is intentionally making an input element with no contrast, but the dark system theme makes the contrast 'good' again, and my extension sees that there is no problem. I see a couple of options of what I can do for a fix:
- Don't do the good contrast check on inputs. Since they don't inherit colors from their parents, it's less likely that the extension will break something by unnecessarily overriding a color. I just need to make sure that this doesn't cause issues in other scenarios.
- Make it so that when using the 'Toggle standard colors' in the extension (via the toolbar button or the site-specific overrides), it also forces all input elements to a default black-on-white color state. Then one would simply add 'docs.google.com' to the 'force standard colors' section in the options.

In the meantime, if you have an addon that allows injection of custom CSS (like Stylish or dotjs), you can fix it with the following rule:
```css
.docs-title-input-label {
    background-color: white;
}
```

**Comment by arigit** [comment]
Thanks Dmitri for the workaround / tip. I do use stylish and added this rule, it  improved the situation a lot. 


```
@namespace url(http://www.w3.org/1999/xhtml);
@-moz-document url-prefix("https://docs.google.")
{
  
.docs-title-input-label {
    background-color: white;
}
}


```

**Comment by DmitriK** [comment]
Release 1.4.0 is available on AMO and should fix this issue by not doing the contrast check for input elements.

**Comment by arigit** [comment]
I installed and tested 1.4b6 that appears to contain the fix 

https://github.com/DmitriK/darkContrast/commit/4834794dcd7d387b591de7b255de0d713bbd1a9a

but it didnt seem to work. This is with FF 54, compact dark theme (shipped by default in firefox), Fedora 25/wayland with adwaita-dark
This the fix work for you in GNOME/adwaita dark?

**Comment by DmitriK** [comment]
KDE and Arc-dark in my case. Either way, I was a bit premature in closing this, since 1.4 is going to be in review for a bit. I'll try testing it on some other systems in the meantime.

**Comment by DmitriK** [comment]
tl:dr; Issue cannot be fixed without breaking other sites. Google is intentionally using white-on-white text, and extension has no way to know it's intentional, so tries to fix the element by default.

The original fix (not doing contrast checks on inputs) caused #22, since Adwaita-dark uses pure white as it's normal input foreground color. When an author intentionally uses the same color (such as the site mentioned in that bug), the extension has no way to tell the difference between "the foreground is white because the site CSS set it that way" and "the foreground is white because the system theme is making it white". This results in the extension mistakenly thinking the foreground color is unset, and trying to fix it by changing the colors (incorrectly, because the author has intentionally made the input element have a dark theme). The same issue can happen with background colors; it's caused any time the author coincidentally uses the same colors as the system theme. The contrast check is thus required to avoid this problem (element already has good contrast, no need to change colors). 

This comes full circle with the google docs site, which is intentionally using white text (defined in CSS) on a white background (assumed to be the default for inputs). The extension can't tell that this situation is intentional, and will try to fix it by default.

I attempted to at least make the "Use standard colors" option force inputs to a white background, but it doesn't work either, because Firefox will buggily still use the theme's background color (and background color only) despite the existence of the user stylesheet (user stylesheet is required to prevent overwriting author styles).

Closing this issue until such a time that Firefox allows me to reliably tell if a style is caused by author CSS or the default browser style. Until then, the custom CSS override in older comments is the only option.

**Comment by arigit** [comment]
Just for the record, a tiny improvement to the stylish rule and make it more seamless.

@namespace url(http://www.w3.org/1999/xhtml);
@-moz-document url-prefix("https://docs.google.")
{  
.docs-title-input-label {
    background-color: white;
}
    
.docs-title-input {
    background-color: white;   
}   
}
